Introduce cgltf_write.h

prideout opened the pull request:
This writes out nicely-formatted JSON and ignores externally-referenced
buffer data.

Does not yet support sparse accessors, skinning, or morph targets, but
it's a good start and I've been using it successfully for a couple weeks
using several test models.

Fixes #52

jkuhlmann commented:
Looking great!

So, just to be clear: Are you planning to fill in the TODOs?

Otherwise, I think it would make sense to return some kind of default value in the conversion functions (e.g. `cgltf_str_from_alpha_mode()`) so that, if the incoming type is something invalid, we don't just return garbage. Or am I missing something?

prideout commented:
Ah, I am planning to fill the TODOs, I thought it would be good make a PR before it's 100% complete to make it easier to iterate and gather feedback.

Given what you're saying we should probably wait to merge, until I finish up all the TODO's.

jkuhlmann commented:
Sorry, that's not how I meant it. :-) I just wanted to know if you were planning to do finish it up or not. Your contribution is of course appreciated either way!
So, yeah, in my opinion we also merge it with the TODOs. I'd just change the conversion functions as mentioned above.

prideout commented:
Understood!  I added a new comment that implements a couple TODOs and adds fallbacks to the conversion functions as you suggested.  Also added a very basic smoke test.

prideout commented:
@jkuhlmann, see latest commit.  Any other requests?

jkuhlmann commented:
@prideout Sorry, had some very busy days. Will get on this the latest on Monday!

jkuhlmann commented:
Great work!
It seems a bit weird that my name is on the copyright; how is that normally handled in Open Source scenarios?

prideout commented:
Good point Johannes, I added a commit that updates the author and year in the copyright header.
